lang/c++/include/avro/NodeImpl.hh

```diff
@@ -235,9 +235,9 @@ using NodeImplSymbolic = NodeImpl<HasName, NoLeaves, NoLeafNames, NoAttributes,
 using NodeImplRecord = NodeImpl<HasName, MultiLeaves, LeafNames, MultiAttributes, NoSize>;
 using NodeImplEnum = NodeImpl<HasName, NoLeaves, LeafNames, NoAttributes, NoSize>;
```

Comment:
Why `enum` still does not support custom attributes ?

Reply:
Do you want me to change that as well? Similar questions on `symbolic` and `union`. I don't have strong opinions but the spec should support all of them.

Reply:
AFAIK the spec says that only the `primitive` types could not have attributes (custom or not).
So, I think `enum` should support them too, just like `map` and `fixed`.

About `union`s - since their members could be `primitive` I think they should not support custom attributes.

I am not sure what `symbolic` is. Is that a reference to previously defined type ? IMO they should not support custom attributes too. These attributes could be defined in the target (record, enum, fixed).